cumulusci/tasks/salesforce/CreateCommunity.py

```diff
@@ -59,6 +55,27 @@ def _run_task(self):
         if r.status_code != 200:
             raise SalesforceException("Unable to prepare org for Communities")
 
+        if self.options.get("url_path_prefix") is None:
+            self.logger.info("Checking for community without a url path prefix.")
+            community_list = self.sf.restful("connect/communities")["communities"]
+            communities = {c["name"]: c for c in community_list}
+
+            for community_name in communities.keys():
+                if communities[community_name].get("urlPathPrefix") is None:
+                    raise SalesforceException(
+                        "A community without a url path prefix already exists, named {}. Try again with the url path prefix".format(
+                            community_name
+                        )
+                    )
+                self.logger.info("No community without a url path prefix found.")
```

Comment:
I checked and in the case where we try to create a 2nd community without a urlPathPrefix, the POST to connect/communities gets this response: `[{'errorCode': 'INVALID_INPUT', 'message': 'Enter a URL for your community.'}]`

So why don't we check for that response after the fact (and raise our more informative error), rather than doing a separate request beforehand. It'll avoid an unnecessary request, and avoid a potential (though unlikely) race condition if a community without a prefix gets created in between the two requests.

Reply:
@davisagli I've added in the tests. After looking at the behavior produced by the api, I am conflicted on whether or not we push our own more helpful error message to supplant that of the API's. It shares the same exception type with the one thrown from the same community name being used. Thus we would need to parse that exception message.

In this case, I feel as though logging a bug with Salesforce that though the message from the API is technically accurate (you do need to enter a url path prefix to get the new community to create), it is less helpful as this is only required because a community without a url path prefix exists.

Reply:
@MatthewBlanski I'm okay with simply showing the error from the server. That has the benefit of being more likely to be something that users will find if they google the error message.